src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php

```diff
@@ -451,14 +451,22 @@ protected function handleView(View $view)
 
     protected function isGrantedOr403($permission)
     {
-        if (!$this->container->has('sylius.authorization_checker')) {
-            return true;
-        }
-
         $permission = $this->config->getPermission($permission);
-
-        if ($permission && !$this->get('sylius.authorization_checker')->isGranted(sprintf('%s.%s.%s', $this->config->getBundlePrefix(), $this->config->getResourceName(), $permission))) {
-            throw new AccessDeniedHttpException();
+        if ($permission) {
+            // Check access in Symfony security, if fail use Sylius one
+            if (!$this->get('security.context')->isGranted($permission)) {
```

Comment:
Would not it be better to remove the sylius.authorization_checker in favor of a Voter, so you can use the security.context for all cases ?